js/classes/physics_component.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * A physics component for an entity.
+ * 
+ * The physics component is responsible for handling the physics of an entity.
+ * It stores the mass, velocity, and acceleration of the entity.
+ * 
+ * How to use:
+ * - Call updateMovement() in the entity's update() method, after applying all forces and drag.
+ * - Call applyForce(PhysicsComponent.GRAVITY) to apply gravity.
+ * - Call applyDrag(PhysicsComponent.AIR_RESISTANCE) to apply air resistance when in the air.
+ * - Call applyDrag(friction) to apply friction when standing on a block.
```

Comment:
Would it be possible to roll these methods into one so that an entity only has to make one `PhysicsComponent` call in their `update` method?

Reply:
Unfortunately not. `applyForce` will take the place of any changes to velocity we make in the Chad class. Each method simply has to be called in specific situations during the `update` loop.

Reply:
Unfortunately not. `applyForce` will take the place of any changes to velocity we make in the Chad class. Each method simply has to be called in specific situations during the `update` loop.

Reply:
Trust me, I've tried my best to condense this stuff xD

I guess it's just the nature of physics. In general, physics are meant to be closely tied with the entity being affected. However, to reduce duplicate code, I wanted to offload the calculations involved in stuff like force. I guess this is the result.

Reply:
Makes sense, I thought that might be the case. It's really not that bad as it is, so no worries. 

> I wanted to offload the calculations involved in stuff like force to another class.
This is one reason I really like this class. We were doing a lot of identical operations in many different places, and although they were simple, putting them in one place is much, much better.